beta-operator-rewards/rewards-requirements.ts

```diff
@@ -0,0 +1,478 @@
+      !beaconAuthorization.isZero()
+    );
+
+    /// tBTC application authorized requirement
+    const tbtcAuthorization = await getAuthorization(
+      walletRegistry,
+      intervalEvents,
+      postIntervalIncreasedEvents.concat(postIntervalDecreasedEvents),
+      stakingProvider,
+      startRewardsBlock,
+      endRewardsBlock
+    );
+
+    authorizations.set(TBTC_AUTHORIZATION, tbtcAuthorization.toString());
+    requirements.set(IS_TBTC_AUTHORIZED_FACTOR, !tbtcAuthorization.isZero());
+
+    /// Start assembling peer data and weighted authorizations
+    peerData[stakingProvider] = {
+      applications: Object.fromEntries(authorizations),
+      requirements: Object.fromEntries(requirements),
+    };
+
+    // TODO: iterate over instances for a given operator address
+    // Assemble
+    //   "instances":[
+    //     {
+    //        "uptimePercent":4.251766217084136,
+    //        "preParams":132,
+    //        "version":"2.0.0-1m",
+    //        "ip":"10.102.0.30:9701"
+    //     },
+    //    (...)
+    //    ],
+
+    peersData.push(peerData);
+  }
+
+  console.log("peersData: ", JSON.stringify(peersData, null, 2));
+
+  // TODO: calculate weighted authorization for a given staking provider
+  // <staking provider address>: {
+  //   beneficiary: <address>,
+  //   weightedAuthorization: <amount> // uptimeCoeficient * min(beaconAuthorization, tbtcAuthorization)
+  // }
+
+  // TODO: Save to file
+  // - all requirements
+  // - weighted authorization for NU team
+  // const jsonObject = await convertToJSON(peersData);
+  // fs.writeFileSync(peersDataFile, JSON.stringify(jsonObject, null, 2));
+}
+
+async function getAuthorization(
+  application: Contract,
+  intervalEvents: any[],
+  postEvents: any[],
+  stakingProvider: string,
+  startRewardsBlock: number,
+  endRewardsBlock: number
+) {
+  // When there were no events during the rewards interval, then we fetch the
+  // authorization after the interval which was the same as for the actual
+  // rewards interval.
+  if (intervalEvents.length == 0) {
+    // Events that were emitted between the [end:firstEventDate|currentDate] dates.
+    // This is used to fetch the authorization that was allocated during the rewards
+    // interval.
+    const postIntervalEvents = await filterEventsByStakingProvider(
+      postEvents,
+      stakingProvider
+    );
+
+    return await authorizationPostRewardsInterval(
+      postIntervalEvents,
+      application,
+      stakingProvider
+    );
+  }
+
+  // There is at least one event emitted during the rewards interval
+  const applicationEvents = intervalEvents.filter((obj) => {
+    return obj.args.application == application.address;
+  });
+
+  return await authorizationForRewardsInterval(
+    applicationEvents,
+    startRewardsBlock,
+    endRewardsBlock
+  );
+}
+
+async function filterEventsByApplications(events: any[]) {
+  const filteredEvents = new Array();
+  for (let j = 0; j < events.length; j++) {
```

Comment:
I think we can condense this with the built in `filter` function:

```js
return events.filter(event => {
  return event.args.application == RandomBeaconAddress ||
    event.args.application == WalletRegistryAddress
})
```

Reply:
Agree! [1e4b765](https://github.com/keep-network/keep-core/pull/3346/commits/1e4b765b1cd76d693ca127134e68948300e4a0e7)